nixos/modules/system/boot/loader/systemd-boot/systemd-boot-builder.py

```diff
@@ -222,9 +222,12 @@ def main():
         gens += get_generations(profile)
     remove_old_entries(gens)
     for gen in gens:
-        write_entry(*gen, machine_id)
-        if os.readlink(system_dir(*gen)) == args.default_config:
-            write_loader_conf(*gen)
+        try:
+            write_entry(*gen, machine_id)
+            if os.readlink(system_dir(*gen)) == args.default_config:
+                write_loader_conf(*gen)
+        except OSError as e:
+            print("ignoring profile '{}' in the list of boot entries because of the following error:\n{}".format(profile, e), file=sys.stderr)
```

Comment:
I don't think this will ever work, because `profile` is undefined at this point...

This should probably be something like:

```suggestion
            print("ignoring profile '{}' in the list of boot entries because of the following error:\n{}".format(gen[0] if gen[0] else "system", e), file=sys.stderr)
```

Fixed in https://github.com/NixOS/nixpkgs/pull/132688.